model/docs/extending_pyrenew.qmd

```diff
@@ -0,0 +1,256 @@
+from pyrenew.process import RtRandomWalkProcess
+from pyrenew.metaclass import DistributionalRV
+```
+
+The following code-chunk defines the model components. Notice that for both the generation interval and the infection feedback, we use a deterministic PMF with equal probabilities:
+
+```{python}
+#| label: model-components
+gen_int = DeterministicPMF(jnp.array([0.25, 0.5, 0.15, 0.1]))
+feedback_strength = DeterministicVariable(0.05)
+
+I0 = DistributionalRV(dist=dist.LogNormal(0, 1), name="I0")
+
+latent_infections = InfectionsWithFeedback(
+  infection_feedback_strength = feedback_strength,
+  infection_feedback_pmf = gen_int,
+)
+
+rt = RtRandomWalkProcess()
+```
+
+With all the components defined, we can build the model:
+
+```{python}
+#| label: build1
+model0 = RtInfectionsRenewalModel(
+    gen_int=gen_int,
+    I0=I0,
+    latent_infections=latent_infections,
+    Rt_process=rt,
+    observation_process=None,
+)
+```
+
+And simulate it from:
+
+```{python}
+#| label: simulate1
+# Sampling and fitting model 0 (with no obs for infections)
+np.random.seed(223)
+with npro.handlers.seed(rng_seed=np.random.randint(1, 600)):
+    model0_samp = model0.sample(n_timepoints=30)
+```
+
+```{python}
+#| label: fig-simulate1
+#| fig-cap: Simulated infections with no observation process
+import matplotlib.pyplot as plt
+fig, ax = plt.subplots()
+ax.plot(model0_samp.latent_infections)
+ax.set_xlabel("Time")
+ax.set_ylabel("Infections")
+plt.show()
+```
+
+## Pyrenew's random variable class
+
+### Fundamentals
+
+All instances of PyRenew's `RandomVariable` should have at least three functions: `__init__()`, `validate()`, and `sample()`. The `__init__()` function is the constructor and initializes the class. The `validate()` function checks if the class is correctly initialized. Finally, the `sample()` method contains the core of the class; it should return a tuple or named tuple. The following is a minimal example of a `RandomVariable` class based on `numpyro.distributions.Normal`:
+
+```python
+from pyrenew.metaclass import RandomVariable
+
+class MyNormal(RandomVariable):
+    def __init__(self, loc, scale):
+        self.validate(scale)
+        self.loc = loc
+        self.scale = scale
+        return None
+
+    @staticmethod
+    def validate(self):
+        if self.scale <= 0:
+            raise ValueError("Scale must be positive")
+        return None
+
+    def sample(self, **kwargs):
+        return (dist.Normal(loc=self.loc, scale=self.scale),)
+```
+
+The `@staticmethod` decorator exposes the `validate` function to be used outside the class. Next, we show how to build a more complex `RandomVariable` class; the `InfectionsWithFeedback` class.
+
+### The `InfectionsWithFeedback` class
+
+Although returning namedtuples are not strictly required, they are recommended as they make the code more readable. The following code-chunk shows how to create a named tuple for the `InfectionsWithFeedback` class:
```

Comment:
```suggestion
Although returning namedtuples is not strictly required, they are the recommended return type as they make the code more readable. The following code-chunk shows how to create a named tuple for the `InfectionsWithFeedback` class:
```

Reply:
Not essential but feel free to commit and re-request review if you like @damonbayer